test/common/router/upstream_request_test.cc

```diff
@@ -32,6 +32,29 @@ class UpstreamRequestTest : public testing::Test {
                                     false, true};
 };
 
+TEST_F(UpstreamRequestTest, CopyDynamicMetaData) {
```

Comment:
I think we also need some integration test here? Maybe find some integration test in `test/integration` to build from?

Reply:
Now there is no filter which will really add the metadata in new namespace. It seems that we cannot add integration test here unless I change the existing network filter's code or add a new network filter which is not in the scope of this PR.